src/wallet/walletutil.cpp

```diff
@@ -61,9 +62,16 @@ std::vector<fs::path> ListWalletDir()
     std::vector<fs::path> paths;
     boost::system::error_code ec;
 
-    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {
+    auto it = fs::recursive_directory_iterator(wallet_dir, ec);
+    if (ec) {
+        LogPrintf("%s: iterator: %s %s\n", __func__, ec.message(), it->path().string());
+        return paths;
+    }
```

Comment:
Wondering what conditions could cause this error. I've read the relevant Boost docs, and tried different ill filesystem setups but without success.